### ISSUE TITLE
Creating a module.

### ISSUE THREAD

**Issue by bitmammoth** [ioannis-mylonas/gdnative-argon2-wrapper#1]
Hi I have found this extremely useful and I was trying to convert this code into a module but I have had a lot of trouble with the templating. Is it possible you would make a module release of the argon2 wrapper? If it could compile into the engine I think it would be great for many people.

Thank you

**Comment by ioannis-mylonas** [comment]
Hello there!  
  
I've created a module branch, for those who prefer using it as a module rather than a GDNative script, along with some basic docs for it. If you prefer using what you were trying to do before (due to customizations or otherwise), I'm not incredibly familiar with Godot modules, but it would seem to me the problem you were facing may be due to the parameter limit of 5 in subclasses, mentioned in the docs (the hashing function takes 6 parameters). Just add `#include "core/method_bind_ext.gen.inc"` and it should compile as expected.  
  
If you're using the one from the `module` branch instead, I've set the SCsub file to look for the library files in the thirdparty folder.  
  
Hope this helps!
